### PR TITLE
Add adjustable zmq port support to satctl

### PR DESCRIPTION
**Pull request by ioztelcan**

### CODE REVIEW

File: src/satctl.c [hunk 1]
```diff
@@ -90,7 +118,7 @@ int main(int argc, char **argv)
 	int csp_version = 2;
 	char * rtable = NULL;
 	char * tun_conf_str = NULL;
-	char * csp_zmqhub_addr[10];
+	char * csp_zmqhub_addr[20];
```
Comment (jenshz) [inline src/satctl.c]: I think 20 is a bit low - maybe just do 128 (you can throw a fully qualified hostname in there).

File: src/satctl.c [hunk 2]
```diff
@@ -245,10 +273,27 @@ int main(int argc, char **argv)
...
 
 	while (csp_zmqhub_idx > 0) {
-		char * zmq_str = csp_zmqhub_addr[--csp_zmqhub_idx];
-		printf("zmq str %s\n", zmq_str);
+		char *zmq_str = csp_zmqhub_addr[--csp_zmqhub_idx];
+		char delimiter[] = ":";
+		int sub_port = CSP_ZMQPROXY_SUBSCRIBE_PORT;
+		int pub_port = CSP_ZMQPROXY_PUBLISH_PORT;
+
+		/* This is how strtok works, each step after first strips the next port number */
```
Comment (jenshz) [inline src/satctl.c]: I think people can read the strtok manual if they want to know how it works.
You could document what the intention is however, e.g.: The first invocation of strtok extracts the hostname. The two following invocations extract the port numbers. Note that we specify NULL to continue with the string specified by the first invocation.

File: src/satctl.c [hunk 3]
```diff
@@ -57,21 +57,49 @@ void usage(void)
...
-	printf(" -r UDP_CONFIG\tUDP configuration string, encapsulate in brackets: \"<lport> <peer ip> <rport>\" (supports multiple) \n");
-	printf(" -z ZMQ_IP\tIP of zmqproxy node (supports multiple)\n");
-	printf(" -p\t\tSetup prometheus node\n");
-	printf(" -R RTABLE\tOverride rtable with this string\n");
-	printf(" -h\t\tPrint this help and exit\n");
+	printf(" -c INTERFACE,\t\t\tUse INTERFACE as CAN interface\n");
+	printf(" -u INTERFACE,\t\t\tUse INTERFACE as UART interface\n");
+	printf(" -b BAUD,\t\t\tUART buad rate\n");
+	printf(" -n NODE\t\t\tUse NODE as own CSP address\n");
+	printf(" -r UDP_CONFIG\t\t\tUDP configuration string, encapsulate in brackets: \"<lport> <peer ip> <rport>\" (supports multiple) \n");
```
Comment (jenshz) [inline src/satctl.c]: Maybe we should use the same syntax as here (or modify this syntax) - now we have two in the same program.

**Review by jenshz — COMMENTED** [review]
Looks good - have a look at my comments, maybe there's something in there.

File: src/satctl.c (hunk 3 above)
Comment (ioztelcan) [reply src/satctl.c]: Which syntax do you mean? I didn't get it.

Comment (jenshz) [reply src/satctl.c]: Just disregard this.

Comment (ioztelcan) [reply src/satctl.c]: Ah do you mean the UDP config? Hmm I can take a look at how it was parsed.

Comment (jenshz) [reply src/satctl.c]: Just update the help message and it should be fine.

**Pull request merged**